context/AppDataContext.tsx

```diff
@@ -62,13 +69,38 @@ const AppDataProvider = ({ children }: { children: React.ReactNode }) => {
     setRepositories(updatedRepositories);
   };
 
+  const filterRepositoriesByTag = (tag: string) => {
```

Comment:
Instead of having `filterRepositoriesByTag`, `filterRepositoriesByLanguage`, `filterRepositoriesByQuery` we should include all apply all these filters when they are set instead of calling the functions directly. As it is now, if a user is on the "golang" language page and starts searching, instead of filtering the "go" results further, they would be filtering all repositories.